homeassistant/components/wiz_light/light.py

```diff
@@ -0,0 +1,428 @@
+
+    @property
+    def max_mireds(self):
+        """Return the warmest color_temp that this light supports."""
+        if self._bulbtype is None:
+            return color_utils.color_temperature_kelvin_to_mired(2200)
+
+        try:
+            return color_utils.color_temperature_kelvin_to_mired(
+                self._bulbtype.kelvin_range.min
+            )
+        except WizLightNotKnownBulb:
+            _LOGGER.info("Kelvin is not present in the library. Fallback to 2200")
+            return color_utils.color_temperature_kelvin_to_mired(2200)
+
+    @property
+    def supported_features(self) -> int:
+        """Flag supported features."""
+        if self._bulbtype:
+            return self.featuremap()
+        # fallback
+        return SUPPORT_FEATURES_RGB
+
+    @property
+    def effect(self):
+        """Return the current effect."""
+        return self._effect
+
+    @property
+    def effect_list(self):
+        """Return the list of supported effects.
+
+        URL: https://docs.pro.wizconnected.com/#light-modes
+        """
+        return self._scenes
+
+    @property
+    def available(self):
+        """Return if light is available."""
+        return self._available
+
+    async def async_update(self):
+        """Fetch new state data for this light."""
+        await self.update_state()
+
+        if self._state is not None and self._state is not False:
+            self.update_brightness()
+            self.update_temperature()
+            self.update_color()
+            self.update_effect()
+            await self.update_scene_list()
+
+    @property
+    def device_info(self):
+        """Get device specific attributes."""
+        _LOGGER.debug(
+            "[wizlight %s] Call device info: MAC: %s - Name: %s - Type: %s",
+            self._light.ip,
+            self._mac,
+            self._name,
+            self._bulbtype.name,
+        )
+        return {
+            "identifiers": {(DOMAIN, self._mac)},
+            "name": self._name,
+            "manufacturer": "WiZ Light Platform",
+            "model": f"{self._bulbtype.name} Mac: {self._mac}",
+        }
+
+    def update_state_available(self):
+        """Update the state if bulb is available."""
+        self._state = self._light.status
+        self._available = True
+
+    def update_state_unavailable(self):
+        """Update the state if bulb is unavailable."""
+        self._state = False
+        self._available = False
+
+    async def update_state(self):
+        """Update the state."""
+        try:
+            await self._light.updateState()
+            if self._light.state is None:
+                self.update_state_unavailable()
+            else:
+                self.update_state_available()
+                # Update the rest of the missing info if available
+                if self._mac is None or self._bulbtype is None:
+                    await self.get_bulb_type()
+                    await self.get_mac()
```

Comment:
Do this in `async_setup_entry` in `__init__`. 

If you can't get it there, then raise `ConfigEntryNotReady` so Home Assistant retries to setup the bulb later.

This will probably help with or solve https://github.com/sbidy/wiz_light/issues/98

Reply:
You mean with __init__ the class init or the `__init__.py` for the integration?

Reply:
Looks like you have it in the expected place. 👍 
 [`b8743a6` (#44779)](https://github.com/home-assistant/core/pull/44779/commits/b8743a68c489f268d91f63963d4172cab3951211)